--- src64/headers/parigen.h	2020-03-05 12:55:11.290389900 -0600
+++ patched_src64/headers/parigen.h	2020-03-05 12:45:50.048980800 -0600
@@ -15,7 +15,11 @@
 
 #ifdef _WIN64
 typedef unsigned long long pari_ulong;
-#define long long long
+#include <limits.h>
+typedef long long pari_long;
+#define YYPTRDIFF_T pari_long
+#define YYPTRDIFF_MAXIMUM LLONG_MAX
+#define long pari_long
 #define labs llabs
 #else
 typedef unsigned long pari_ulong;
