diff -u config/arch-osname config.new/arch-osname
--- config/arch-osname	2022-09-19 10:42:30.000000000 -0500
+++ config.new/arch-osname	2022-11-13 07:00:04.071022300 -0600
@@ -5,7 +5,8 @@
 if test -n "$myuname"; then
   myuname=`echo $myuname | sed -e 's/^[^=]*=//' -e 's,/,,g' | \
 	  tr '[A-Z]' '[a-z]' | tr '\012' ' '`
-  set X $myuname; shift; osname=$1
+  #set X $myuname; shift; osname=$1
+  osname="mingw"
   case "$osname" in
   irix*)    osname=irix;;
   fx2800)   arch=fx2800; osname=concentrix;;
diff -u config/get_gmp config.new/get_gmp
--- config/get_gmp	2020-10-26 07:43:04.000000000 -0500
+++ config.new/get_gmp	2022-11-13 07:00:04.071022300 -0600
@@ -40,9 +40,10 @@
   fi
 fi
 
-if test -r $exe; then
-  gmp_version=`env LD_LIBRARY_PATH="$LD_LIBRARY_PATH$dir_sep$gmp" $RUNTEST $exe`;
-fi
+gmp_version="6.1.2"
+#if test -r $exe; then
+#  gmp_version=`env LD_LIBRARY_PATH="$LD_LIBRARY_PATH$dir_sep$gmp" $RUNTEST $exe`;
+#fi
 case "$gmp_version" in
   unsupported) gmp=
     echo "### Your GMP library ABI is unsupported.";;
@@ -57,14 +58,14 @@
   *) if test -z "$gmp"; then gmp=yes; fi;;
 esac
 
-if test -n "$gmp"; then
-  cmd="$CC $CFLAGS $extraflag $GMPINCLUDE -o $exe gmp_mismatch.c $GMPLIBS"
-  . log_cmd
-  if test ! -r $exe; then
-    gmp=
-    echo "### GMP headers mismatch: try both --with-gmp-lib and --with-gmp-include"
-  fi
-fi
+#if test -n "$gmp"; then
+#  cmd="$CC $CFLAGS $extraflag $GMPINCLUDE -o $exe gmp_mismatch.c $GMPLIBS"
+#  . log_cmd
+#  if test ! -r $exe; then
+#    gmp=
+#    echo "### GMP headers mismatch: try both --with-gmp-lib and --with-gmp-include"
+#  fi
+#fi
 
 . cleanup_exe
 
