--- libcache/pari/include/pari/paristio.h	2022-11-12 06:43:02.000000000 -0600
+++ libcache/pari/include/pari/paristio.h.new	2022-11-12 06:42:05.000000000 -0600
@@ -14,6 +14,9 @@
 
 /* This file contains memory and I/O management definitions       */
 
+#ifndef PARISTIO_H
+#define PARISTIO_H
+
 typedef struct {
   long s, us;
 } pari_timer;
@@ -402,3 +405,4 @@
 #ifndef SIGINT
 #  define SIGINT 2
 #endif
+#endif
