--- a/src/basemath/trans1.c
+++ b/src/basemath/trans1.c
@@ -2269,8 +2269,10 @@ sin_p(GEN x)
 static GEN
 cxexp(GEN x, long prec)
 {
-  GEN r,p1,p2, y = cgetg(3,t_COMPLEX);
+  GEN r, p1, p2, y = cgetg(3,t_COMPLEX);
   pari_sp av = avma, tetpil;
+  long l;
+  l = precision(x); if (l > prec) prec = l;
   r = gexp(gel(x,1),prec);
   if (gequal0(r)) { gel(y,1) = r; gel(y,2) = r; return y; }
   gsincos(gel(x,2),&p2,&p1,prec);
@@ -2932,6 +2934,7 @@ glog(GEN x, long prec)
     }
     case t_COMPLEX:
       if (ismpzero(gel(x,2))) return glog(gel(x,1), prec);
+      l = precision(x); if (l > prec) prec = l;
       if (ismpzero(gel(x,1)))
       {
         GEN a = gel(x,2), b;
@@ -2940,7 +2943,6 @@ glog(GEN x, long prec)
         a = isint1(a) ? gen_0: glog(a,prec);
         return gerepilecopy(av, mkcomplex(a, b));
       }
-      l = precision(x); if (l > prec) prec = l;
       if (prec >= LOGAGMCX_LIMIT) return logagmcx(x, prec);
       y = cgetg(3,t_COMPLEX);
       gel(y,2) = garg(x,prec);
--- a/src/basemath/trans3.c
+++ b/src/basemath/trans3.c
@@ -2385,7 +2385,8 @@ polylog(long m, GEN x, long prec)
     return gerepileupto(av, gneg(glog(gsub(gen_1,x), prec)));
   }
 
-  l = precision(x); if (!l) l = prec;
+  l = precision(x);
+  if (!l) l = prec; else prec = l;
   res = cgetc(l); av = avma;
   x = gtofp(x, l+EXTRAPRECWORD);
   e = gexpo(gnorm(x));

