diff -u pari_src/config/arch-osname config_new/arch-osname
--- pari_src/config/arch-osname	2011-09-22 15:02:11.000000000 -0500
+++ config_new/arch-osname	2018-04-11 06:35:23.947800500 -0500
@@ -5,7 +5,8 @@
 if test -n "$myuname"; then
   myuname=`echo $myuname | sed -e 's/^[^=]*=//' -e 's,/,,g' | \
 	  tr '[A-Z]' '[a-z]' | tr '\012' ' '`
-  set X $myuname; shift; osname=$1
+  #set X $myuname; shift; osname=$1
+  osname="mingw"
   case "$osname" in
   irix*)    osname=irix;;
   fx2800)   arch=fx2800; osname=concentrix;;
diff -u pari_src/config/get_gmp config_new/get_gmp
--- pari_src/config/get_gmp	2017-01-14 06:26:41.000000000 -0600
+++ config_new/get_gmp	2018-04-11 07:20:35.448573300 -0500
@@ -39,9 +39,10 @@
   fi
 fi
 
-if test -r $exe; then
-  gmp_version=`env LD_LIBRARY_PATH="$LD_LIBRARY_PATH$dir_sep$gmp" $RUNTEST $exe`;
-fi
+gmp_version="6.1.2"
+#if test -r $exe; then
+#  gmp_version=`env LD_LIBRARY_PATH="$LD_LIBRARY_PATH$dir_sep$gmp" $RUNTEST $exe`;
+#fi
 case "$gmp_version" in
   unsupported) gmp=
     echo "### Your GMP library ABI is unsupported.";;
@@ -56,14 +57,14 @@
   *) if test -z "$gmp"; then gmp=yes; fi;;
 esac
 
-if test -n "$gmp"; then
-  cmd="$CC $CFLAGS $extraflag $GMPINCLUDE -o $exe gmp_mismatch.c $GMPLIBS"
-  . log_cmd
-  if test ! -r $exe; then
-    gmp=
-    echo "### GMP headers mismatch: try both --with-gmp-lib and --with-gmp-include"
-  fi
-fi
+#if test -n "$gmp"; then
+#  cmd="$CC $CFLAGS $extraflag $GMPINCLUDE -o $exe gmp_mismatch.c $GMPLIBS"
+#  . log_cmd
+#  if test ! -r $exe; then
+#    gmp=
+#    echo "### GMP headers mismatch: try both --with-gmp-lib and --with-gmp-include"
+#  fi
+#fi
 
 . cleanup_exe
 
